extension-manifest-v3/src/utils/engines.js

```diff
@@ -378,6 +393,7 @@ export async function createCustomEngine(filters = '') {
     enableHtmlFiltering: true,
   });
   const engine = FiltersEngine.parse(filters, config);
+  engine.updateEnv(ENV);
```

Comment:
I would create another `parseEngine` function to use the same pattern, that updateEnv is not used directly.